app/models/purchase.rb

```diff
@@ -3821,8 +3835,4 @@ def trigger_iffy_moderation
         Iffy::Product::IngestJob.perform_async(link.id)
       end
     end
-
-    def fetch_installment_plan
-      installment_plan || subscription&.last_payment_option&.installment_plan
```

Comment:
`last_payment_option` will point to the new `ProductInstallmentPlan` if there is a change in number of installments (3 -> 5) which caused the bug